jovian/utils/clone.py

```diff
@@ -77,10 +77,11 @@ def clone(slug, version=None, fresh=True, include_outputs=True):
     # If fresh clone, create directory
     if fresh:
         if os.path.exists(title):
-            i = 1
-            while os.path.exists(title + '-' + str(i)):
-                i += 1
-            title = title + '-' + str(i)
+            if not overwrite:
```

Comment:
nested `if`s often lead to bugs and make it harder to understand the code. Can we refactor this to remove nesting?

There are 4 cases here:
- fresh & doesn't already exist (overwrite doesn't matter): 
- fresh & exists & overwrite
- fresh & exists & doesn't overwrite
- not fresh (existence & overwrite don't matter)

Would be best to flatten out to these 4 conditions, and refactor any repeated code into helper functions.

Reply:
Done, please review again :)